blockchain/dkg/utilities.go

```diff
@@ -127,3 +126,108 @@ func LogReturnErrorf(logger *logrus.Entry, mess string, args ...interface{}) err
 }
+
+// GetValidatorAddressesFromPool retrieves validator addresses from ValidatorPool
+func GetValidatorAddressesFromPool(callOpts *bind.CallOpts, eth interfaces.Ethereum, logger *logrus.Entry) ([]common.Address, error) {
+	c := eth.Contracts()
+
+	addresses, err := c.ValidatorPool().GetValidatorAddresses(callOpts)
+	if err != nil {
+		message := fmt.Sprintf("could not get validator addresses from ValidatorPool: %v", err)
+		logger.Errorf(message)
+		return nil, err
+	}
+
+	return addresses, nil
+}
+
+// ComputeDistributedSharesHash computes the distributed shares hash, encrypted shares hash and commitments hash
+func ComputeDistributedSharesHash(encryptedShares []*big.Int, commitments [][2]*big.Int) ([32]byte, [32]byte, [32]byte, error) {
+	var emptyBytes32 [32]byte
+
+	// encrypted shares hash
+	encryptedSharesBin, err := bn256.MarshalBigIntSlice(encryptedShares)
+	if err != nil {
+		return emptyBytes32, emptyBytes32, emptyBytes32, fmt.Errorf("ComputeDistributedSharesHash encryptedSharesBin failed: %v", err)
+	}
+	hashSlice := crypto.Hasher(encryptedSharesBin)
+	var encryptedSharesHash [32]byte
+	copy(encryptedSharesHash[:], hashSlice)
+
+	// commitments hash
+	commitmentsBin, err := bn256.MarshalG1BigSlice(commitments)
+	if err != nil {
+		return emptyBytes32, emptyBytes32, emptyBytes32, fmt.Errorf("ComputeDistributedSharesHash commitmentsBin failed: %v", err)
+	}
+	hashSlice = crypto.Hasher(commitmentsBin)
+	var commitmentsHash [32]byte
+	copy(commitmentsHash[:], hashSlice)
+
+	// distributed shares hash
+	var distributedSharesBin = append(encryptedSharesHash[:], commitmentsHash[:]...)
+	hashSlice = crypto.Hasher(distributedSharesBin)
+	var distributedSharesHash [32]byte
+	copy(distributedSharesHash[:], hashSlice)
+
+	return distributedSharesHash, encryptedSharesHash, commitmentsHash, nil
+}
+
+func WaitConfirmations(txHash common.Hash, ctx context.Context, logger *logrus.Entry, eth interfaces.Ethereum) error {
+	var done = false
+
+	for !done {
+
+		receipt, err := eth.GetGethClient().TransactionReceipt(ctx, txHash)
+
+		if err != nil {
+			logger.Errorf("waiting for receipt failed: %v", err)
+			return err
+		}
+
+		if receipt == nil {
+			return errors.New("receipt is nil")
+		}
+
+		// Check receipt to confirm we were successful
+		if receipt.Status != uint64(1) {
+			message := fmt.Sprintf("tx status (%v) indicates failure: %v", receipt.Status, receipt.Logs)
+			logger.Error(message)
+			return errors.New(message)
+		}
+
+		receiptBlock := receipt.BlockNumber.Uint64()
+
+		// receipt was successful, let's wait for `nConfirmation` block confirmations
+		currentHeight, err := eth.GetCurrentHeight(ctx)
+		if err != nil {
+			return LogReturnErrorf(logger, "could not get block height: %v", err)
+		}
+
+		if currentHeight >= receiptBlock+eth.GetFinalityDelay() {
+			done = true
+		}
+
+		time.Sleep(5 * time.Second)
+
+	}
+
+	return nil
+}
+
+func IncreaseFeeAndTipCap(gasFeeCap, gasTipCap, percentage *big.Int) (*big.Int, *big.Int) {
```

Comment:
var names misleading, not set 10%.
use mod to write ceil function to not blind increase by 1